lib/lists.nix

```diff
@@ -144,10 +143,9 @@ rec {
     fold' 0;
 
   /**
-    `fold` is an alias of `foldr` for historic reasons
+    `fold` is an alias of `foldr` for historic reasons. Remove in 26.05.
```

Comment:
This is still user facing documentation. And should conform to [documentation conventions](https://github.com/NixOS/nixpkgs/blob/master/doc/README.md#writing-function-documentation)

```suggestion
    `fold` is an alias of `foldr` for historic reasons. 
    
    ::: {.warning}
    This function will be removed in 26.05.
    :::
```